apps/api-extractor/src/api/ExtractorConfig.ts

```diff
@@ -856,42 +888,79 @@ export class ExtractorConfig {
         }
       }
 
-      let apiReportEnabled: boolean = false;
-      let reportFilePath: string = '';
-      let reportTempFilePath: string = '';
-      let apiReportIncludeForgottenExports: boolean = false;
-      if (configObject.apiReport) {
-        apiReportEnabled = !!configObject.apiReport.enabled;
-
-        const reportFilename: string = ExtractorConfig._expandStringWithTokens(
-          'reportFileName',
-          configObject.apiReport.reportFileName || '',
-          tokenContext
-        );
+      const apiReportEnabled: boolean = configObject.apiReport?.enabled ?? false;
+      const apiReportIncludeForgottenExports: boolean =
+        configObject.apiReport?.includeForgottenExports ?? false;
+      let reportDirectoryPath: string = tokenContext.projectFolder;
+      let reportTempDirectoryPath: string = tokenContext.projectFolder;
+      const reportConfigs: IApiReportConfig[] = [];
+      if (apiReportEnabled) {
+        // Undefined case checked above where we assign `apiReportEnabled`
+        const apiReportConfig: IConfigApiReport = configObject.apiReport!;
+
+        const reportFileNameSuffix: string = '.api.md';
+        let reportFileNameBase: string;
+        if (apiReportConfig.reportFileName) {
+          if (
+            apiReportConfig.reportFileName.indexOf('/') >= 0 ||
+            apiReportConfig.reportFileName.indexOf('\\') >= 0
+          ) {
+            throw new Error(
+              `The "reportFileName" setting contains invalid characters: "${apiReportConfig.reportFileName}"`
+            );
+          }
 
-        if (!reportFilename) {
-          // A merged configuration should have this
-          throw new Error('The "reportFilename" setting is missing');
+          const iSuffix: number = apiReportConfig.reportFileName.indexOf(reportFileNameSuffix);
```

Comment:
Rename `iSuffix` to `suffixIndex` following our coding convention that identifiers should be comprised of full English words